860 activity task ssantiago (#1101)
temporalio/documentation: docs/concepts/what-is-an-activity-task.md

id: what-is-an-activity-task
title: What is an Activity Task?
sidebar_label: Activity Task
description: An Activity Task contains the context needed to make an Activity Task Execution.
tags:
  - explanation
---

An Activity Task contains the context needed to proceed with an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).

> 
> Activity Tasks contain Events that are needed to execute the Activity. These events are found within the Activity Task's `Context`.
> 

Reviewer: ```suggestion

```
Just remove this in favor of the sentences below.

id: what-is-an-activity-task
title: What is an Activity Task?
sidebar_label: Activity Task
description: An Activity Task contains the context needed to make an Activity Task Execution.
tags:
  - explanation
---

An Activity Task contains the context needed to proceed with an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).

> From the API, users see an Activity being executed and assume that it runs once.

Reviewer: Not sure this feels like this sentence belongs here...

> Activity Tasks contain Events that are needed to execute the Activity. These events are found within the Activity Task's `Context`.

Reviewer: I would change this to - 
Activity Tasks largely represent the Activity Task Scheduled Event which contains the data needed to execute an Activity Function.
If Heartbeat data is being passed, an Activity Task will also contain the latest Heartbeat details for example.

Something like that maybe

id: what-is-an-activity-task
title: What is an Activity Task?
sidebar_label: Activity Task
description: An Activity Task contains the context needed to make an Activity Task Execution.
tags:
  - explanation
---

An Activity Task contains the context needed to proceed with an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).

> Activity Tasks are meant to run an [Activity Definition](/docs/concepts/what-is-an-activity-definition) once. Should this fail, the Activity Task is retried until it's completed successfully.

Reviewer 2: ```suggestion
Activity Tasks are meant to run an [Activity Definition](/docs/concepts/what-is-an-activity-definition) once.
If an Activity Task Execution fails, the Activity Task is retried in accordance with its Retry Policy.
```

An Activity Task is meant to run an [Activity Definition](/docs/concepts/what-is-an-activity-definition) once.

> If an Activity Task Execution fails, the Activity Task is retried in accordance with its Retry Policy.

Reviewer: I think we can drop this bit of info as it is covered in the Activity Task Execution page.

> Activity Tasks are assigned to an Activity Task Queue upon an [ActivityTaskScheduled Event](/docs/concepts/what-is-an-activity-task#scheduling-activity-tasks). Once a Task is executed, an Activity's Events are run until the Activity successfully completes.

Reviewer 2: ```suggestion
Activity Tasks are assigned to an Activity Task Queue upon an [ActivityTaskScheduled Event](/docs/concepts/what-is-an-activity-task#scheduling-activity-tasks).
```
An Activity doesn't run Events. Events are generated by the Temporal Server.

> Activity Tasks contain Events that are needed to execute the Activity. These events are found within the Activity Task's [Context](/docs/concepts/what-is-an-activity-task#context-events).

Reviewer: Hmm... why does this link right back to this page?
Also, an Activity Task does not contain Events - nor is it an Event.
Events live in the Workflow Execution Event History.

My current suggestion would drop the "Context Events" header and this sentence above and just have a solid set of sentences / paragraph of info.

> Activity Tasks contain Events that contain what's needed to execute the Activity. These events are found within the Activity Task's [Context](/docs/concepts/what-is-an-activity-task#context-events).

Reviewer 2: An Activity Task can contain context, but Events are generated by the Temporal Server.

> The Temporal Cluster puts an Activity Task into the Task Queue. This correlates to the [ActivityTaskScheduled Event](/docs/concepts/what-is-an-activity-task#scheduling-activity-tasks) in the Activity Task's Context.

Reviewer: Each sentence on new line

> Upon completion, the Activity Task responds to the cluster with a specific Event:

Reviewer: Move this into the Activity Task Execution page -

But the Worker responds to the Cluster once the Activity Task Execution finishes (with any of the below Events)

Reviewer: Activity Tasks do not respond.
Workers respond. 
And I think Activity Task Execution is a better spot for this - merge with my suggestion above maybe.

> ActivityTaskScheduled prepares the Activity for execution. ActivityTaskStarted begins the Activity.

Reviewer 2: These Events are generated by the Temporal Server. I'm hesitant to cast them as actors.

> ActivityTaskScheduled prepares the Activity for execution. ActivityTaskStarted begins the Activity.

Reviewer: Ya, these Events are correlations to other things happening - 

The Temporal Cluster puts an Activity Task into the Task Queue and that correlates to the ActivityTaskScheduled Event.

id: what-is-an-activity-task
title: What is an Activity Task?
sidebar_label: Activity Task
description: An Activity Task contains the context needed to make an Activity Task Execution.
tags:
  - explanation
---

An Activity Task contains the context needed to proceed with an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).

Activity Tasks are meant to run an [Activity Definition](/docs/concepts/what-is-an-activity-definition) once.
If an Activity Task Execution fails, the Activity Task is retried in accordance with its Retry Policy.

#### Scheduling Activity Tasks

> Activity Tasks are assigned to an Activity Task Queue upon an [ActivityTaskScheduled Event](/docs/concepts/what-is-an-activity-task#scheduling-activity-tasks).

Reviewer: I think its more of a correlation -
An Activity Task comes into existence when it is scheduled - and the Event is added to the Workflow's Event History.

#### Context Events

Activity Tasks contain Events that are needed to execute the Activity. These events are found within the Activity Task's [Context](/docs/concepts/what-is-an-activity-task#context-events).

[event explanation?]

ActivityTaskScheduled prepares the Activity for execution. ActivityTaskStarted begins the Activity.

> Upon completion, the Activity Task responds to the cluster with an Event status of _success_ or _failure_.

Reviewer: The Worker responds with a specific Event - (one of ActivityTaskCompleted, ActivityTaskFailed, ActivityTaskTimedOut, ActivityTaskCanceled, ActivityTaskedTerminated)

> [failure events may have retry?]

Reviewer: Yes, if the Retry Policy dictates it - and if the Events are either ActivityTaskFailed or ActivityTaskTimedOut for example - then yes, the Cluster would schedule another Activity Task.
